### ISSUE TITLE
grpc-plugin builder failed to create a store

### ISSUE THREAD

**Issue by xk1010xcn** [muhammadn/jaeger-s3#82]
I've tried this plugin, but an error has occurred. I don't know if it's a problem with the Docker mount directory. am i doing something wrong?

**docker command：**
`
docker run --name jaeger_s3 -itd \
-v d/Work/docker/config/jaeger/plugin:/plugin \
-v d/Work/docker/config/jaeger/config:/plugin_config \
-e SPAN_STORAGE_TYPE=grpc-plugin \
-e GRPC_STORAGE_PLUGIN_BINARY=/plugin \
-e GRPC_STORAGE_PLUGIN_CONFIGURATION_FILE=/plugin_config \
jaegertracing/all-in-one:latest
`

**error message：**
{"level":"info","ts":1701347847.8502195,"caller":"flags/service.go:117","msg":"Mounting metrics handler on admin server","route":"/metrics"}
{"level":"info","ts":1701347847.8502717,"caller":"flags/service.go:123","msg":"Mounting expvar handler on admin server","route":"/debug/vars"}
{"level":"info","ts":1701347847.8503962,"caller":"flags/admin.go:104","msg":"Mounting health check on admin server","route":"/"}
{"level":"info","ts":1701347847.850435,"caller":"flags/admin.go:115","msg":"Starting admin HTTP server","http-addr":":14269"}
{"level":"info","ts":1701347847.8504624,"caller":"flags/admin.go:96","msg":"Admin server started","http.host-port":"[::]:14269","health-status":"unavailable"}
{"level":"fatal","ts":1701347847.8574052,"caller":"./main.go:107","msg":"Failed to init storage factory","error":"grpc-plugin builder failed to create a store: error attempting to connect to plugin rpc client: fork/exec /plugin: permission denied","stacktrace":"main.main.func1\n\t./main.go:107\ngithub.com/spf13/cobra.(*Command).execute\n\tgithub.com/spf13/cobra@v1.3.0/command.go:856\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\tgithub.com/spf13/cobra@v1.3.0/command.go:974\ngithub.com/spf13/cobra.(*Command).Execute\n\tgithub.com/spf13/cobra@v1.3.0/command.go:902\nmain.main\n\t./main.go:226\nruntime.main\n\truntime/proc.go:255"}

**Comment by muhammadn** [comment]
@xk1010xcn Try this:

```
GOOS=linux GOARCH=amd64 go build ./cmd/jaeger-objectstorage
```
then

``` 
docker run --name jaeger -e SPAN_STORAGE_TYPE=grpc-plugin  -e GRPC_STORAGE_PLUGIN_BINARY="/app/jaeger-objectstorage"  -e GRPC_STORAGE_PLUGIN_CONFIGURATION_FILE=/app/config-example.yaml  -e GRPC_STORAGE_PLUGIN_LOG_LEVEL=DEBUG --mount type=bind,source="$(pwd)",target=/app  -e COLLECTOR_ZIPKIN_HOST_PORT=:9411 -e ASSUME_NO_MOVING_GC_UNSAFE_RISK_IT_WITH=go1.19  -p 5775:5775/udp  -p 6831:6831/udp  -p 6832:6832  -p 5778:5778  -p 16686:16686  -p 14268:14268 -p 14250:14250  -p 9411:9411  jaegertracing/all-in-one:1.22
```

**Comment by muhammadn** [comment]
@xk1010xcn We will be releasing a new version soon but with a breaking change to how we store the traces. 

We will update it here:

https://community.flitnetics.com/t/upcoming-update-of-jaeger-objectstorage/9

**Comment by muhammadn** [comment]
@xk1010xcn 
I've released version v3.0.0 of this plugin. v3 uses tempo backend to store in GCS/AWS S3/AzureBlob instead of Loki since Tempo backend has matured so jaeger-objectstorage reads the traces from Tempo, so this would solve all your issues.   🎉

Please try v3.0.0 out and tell me how it goes.

Closing for now. Do comment if you feel i'm missing something.
